/*
 * Aspia: Remote desktop and file transfer tool.
 * Copyright (C) 2018 Dmitry Chapyshev <dmitry@aspia.ru>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

import "authorization.proto";

package aspia.proto.notifier;

message Session
{
    string uuid                   = 1;
    string remote_address         = 2;
    string username               = 3;
    auth.SessionType session_type = 4;
}

message SessionClose
{
    string uuid = 1;
}

message KillSession
{
    string uuid = 1;
}

message NotifierToService
{
    KillSession kill_session = 1;
}

message ServiceToNotifier
{
    Session session            = 1;
    SessionClose session_close = 2;
}
